internal/interceptor.go

```diff
@@ -351,7 +351,7 @@ type ClientUpdateWorkflowInput struct {
 	Args                []interface{}
 	RunID               string
 	FirstExecutionRunID string
-	WaitPolicy          *updatepb.WaitPolicy
+	LifeCycleWaitStage  UpdateLifeCycleStage
```

Comment:
Not sure we need to add `LifeCycle` in the field name. Just `WaitStage` or maybe `WaitForStage` makes sense to me